File: pythonlib/winfox/rdp/dialog.py
from typing import Any, Dict, Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from camoufox.rdp_api import RDPPage


class RDPDialog:
    def __init__(
        self,
        page: "RDPPage",
        dialog_id: int,
        dialog_type: str,
        message: str,
        default_value: Optional[str] = None,
    ):
        self._page = page
        self._id = dialog_id
        self.type = dialog_type
        self.message = message
        self.default_value = default_value
        self.handled = False
        self.accepted: Optional[bool] = None
        self.prompt_text: Optional[str] = None

    def _update_from_state(self, state: Dict[str, Any]) -> None:
        self.handled = bool(state.get("handled", self.handled))
        self.accepted = state.get("accepted", self.accepted)
        self.prompt_text = state.get("promptText", self.prompt_text)

    async def accept(self, prompt_text: Optional[str] = None) -> None:
        if self.handled:
            return
        state = await self._page._resolve_dialog(self._id, accepted=True, prompt_text=prompt_text)
        if isinstance(state, dict):
            self._update_from_state(state)
        else:
            self.handled = True
            self.accepted = True
            self.prompt_text = prompt_text

    async def dismiss(self) -> None:
        if self.handled:
            return
        state = await self._page._resolve_dialog(self._id, accepted=False, prompt_text=None)
        if isinstance(state, dict):
            self._update_from_state(state)
        else:
            self.handled = True
            self.accepted = False
            self.prompt_text = None

File: packages/winfox/src/rdp/dialog.ts
import type { RDPPage } from "./page.js";

export class RDPDialog {
  private _page: RDPPage;
  private _id: number;
  public type: string;
  public message: string;
  public defaultValue: string | null;
  public handled: boolean = false;
  public accepted: boolean | null = null;
  public promptText: string | null = null;

  constructor(page: RDPPage, dialogId: number, dialogType: string, message: string, defaultValue: string | null = null) {
    this._page = page;
    this._id = dialogId;
    this.type = dialogType;
    this.message = message;
    this.defaultValue = defaultValue;
  }

  private _updateFromState(state: any) {
    this.handled = state.handled ?? this.handled;
    this.accepted = state.accepted ?? this.accepted;
    this.promptText = state.promptText ?? this.promptText;
  }

  async accept(promptText: string | null = null): Promise<void> {
    if (this.handled) return;
    const state = await this._page._resolveDialog(this._id, true, promptText);
    if (state && typeof state === 'object') {
      this._updateFromState(state);
    } else {
      this.handled = true;
      this.accepted = true;
      this.promptText = promptText;
    }
  }

  async dismiss(): Promise<void> {
    if (this.handled) return;
    const state = await this._page._resolveDialog(this._id, false, null);
    if (state && typeof state === 'object') {
      this._updateFromState(state);
    } else {
      this.handled = true;
      this.accepted = false;
      this.promptText = null;
    }
  }
}
